tests/vxlan/test_vnet_route_leak.py

```diff
@@ -0,0 +1,180 @@
+    generate_dut_config_files(duthost, minigraph_facts, vnet_test_params, vnet_config)
+    apply_dut_config_files(duthost, vnet_test_params)
+
+    # In this case yield is used only to separate this fixture into setup and teardown portions
+    yield None
+
+    if vnet_test_params[CLEANUP_KEY]:
+        cleanup_vnet_routes(duthost, vnet_test_params)
+        cleanup_dut_vnets(duthost, minigraph_facts, vnet_config)
+        cleanup_vxlan_tunnels(duthost, vnet_test_params)
+
+        logger.info("Reloading CONFIG_DB from JSON")
+
+        duthost.shell(CONFIG_SAVE_CMD)
+        duthost.shell(RESTART_BGP_CMD)
+    else:
+        logger.info("Skipping cleanup")
+
+
+def bgp_connected(duthost):
+    """
+    Checks if BGP connections are up
+
+    BGP connections are "up" once they have received all prefixes (6400) from all neighbors
+
+    Args:
+        duthost: DUT host object
+    """
+
+    bgp_summary = duthost.shell(SHOW_BGP_SUMMARY_CMD)["stdout"].split("\n")
+    logger.debug("BGP Summary: {}".format(bgp_summary))
+
+    # Match entire line containing BGP neighbor information
+    bgp_neighbor_regex = re.compile(r"^([0-9]{1,3}\.){3}[0-9]{1,3}.*")
+
+    bgp_neighbors = []
+
+    for line in bgp_summary:
+        matched = bgp_neighbor_regex.match(line)
+
+        if matched:
+            bgp_neighbors.append(str(matched.group(0)))
+
+    if not bgp_neighbors:
+        return False
+
+    for neighbor in bgp_neighbors:
+        if neighbor.split()[9] != "6400":
+            return False
+
+    logger.info("BGP sessions up")
+
+    return True
+
+
+def no_vnet_route_leak(duthost):
+    """
+    Gets all VNET routes and checks that they are not advertised to any BGP neighbors
+
+    Args:
+        duthost: DUT host object
+    """
+
+    vnet_routes = duthost.shell(SHOW_VNET_ROUTES_CMD)["stdout"].split("\n")
+    logger.debug("VNET prefixes: {}".format(vnet_routes))
+
+    vnet_prefixes = []
+
+    for line in vnet_routes:
+        if any(char.isdigit() for char in line):
+            vnet_prefixes.append(line.split()[1])
+
+    bgp_summary = duthost.shell(SHOW_BGP_SUMMARY_CMD)["stdout"].split("\n")
+    logger.debug("BGP Summary: {}".format(bgp_summary))
+
+    # Match only IP addresses at the beginning of the line
+    # Only IP addresses of neighbors should be matched by this
+    bgp_neighbor_addr_regex = re.compile(r"^([0-9]{1,3}\.){3}[0-9]{1,3}")
+
+    bgp_neighbors = []
+
+    for line in bgp_summary:
+        matched = bgp_neighbor_addr_regex.match(line)
+
+        if matched:
+            bgp_neighbors.append(str(matched.group(0)))
+
+    leaked_routes = {}
+
+    for neighbor in bgp_neighbors:
+        adv_routes = duthost.shell(SHOW_BGP_ADV_ROUTES_CMD_TEMPLATE.format(neighbor))["stdout"]
+
+        for prefix in vnet_prefixes:
+            if prefix in adv_routes:
+                if neighbor not in leaked_routes:
+                    leaked_routes[neighbor] = []
+                leaked_routes[neighbor].append(str(prefix))
```

Comment:
You may be able to simplify this by making leaked_routes a `default_dict(list)`

Reply:
I don't remember if that will break the `not leaked_routes` check at the bottom or not. If it does, you can probably leave this part alone.

Reply:
will fix in next push